app/scripts/controllers/transactions/index.js

```diff
@@ -726,7 +727,11 @@ export default class TransactionController extends EventEmitter {
     const txMeta = this.txStateManager.getTransaction(txId);
     // add network/chain id
     const chainId = this.getChainId();
+    const type = isEIP1559Transaction(txMeta)
```

Comment:
One final thing I forgot is that we should honor the type provided by dapps if they set type themselves.

Reply:
oh i see that the type parameter is set *before* the spread of txParams so this is already cared for!